Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/internal/RiskResultDatabase.kt

```diff
@@ -41,9 +41,15 @@ abstract class RiskResultDatabase : RoomDatabase() {
 
     @Dao
     interface RiskResultsDao {
-        @Query("SELECT * FROM riskresults")
+        @Query("SELECT * FROM riskresults ORDER BY calculatedAt DESC")
         fun allEntries(): Flow<List<PersistedRiskLevelResultDao>>
 
+        @Query("SELECT * FROM riskresults ORDER BY calculatedAt DESC LIMIT :limit")
+        fun latestEntries(limit: Int): Flow<List<PersistedRiskLevelResultDao>>
+
+        @Query("SELECT * FROM (SELECT * FROM riskresults ORDER BY calculatedAt DESC LIMIT 1) UNION ALL SELECT * FROM (SELECT * FROM riskresults where failureReason is null ORDER BY calculatedAt DESC LIMIT 1)")
```

Comment:
are the outer `SELECT * FROM` on both sides of  UNION ALL really needed?

Reply:
I would have said "no" before trying, but I can't get it to work without. Room throws errors on compile. @BMItter Any idea?

Reply:
Its a specification for sqlite/room; only possible with subqueries as done.
so all right here